index.js

```diff
@@ -76,7 +77,13 @@ Retry.prototype._try = function () {
     self.retrying = undefined;
     self.abort = undefined;
 
-    return self._try();
+
+    if (self.options.retryLimit > -1 && self.failures >= self.options.retryLimit) {
+      self.end();
+      return Promise.reject(new Error('Retry limit reached'));
+    } else {
```

Comment:
No need to have an `} else {` here – if the first `if` has returned than anything following it won't run anyway so just put the last return at the end of the method.